state/MapSearchProvider/MapSearchProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import { Position } from 'geojson'
+import { createContext, PropsWithChildren, useCallback, useState } from 'react'
+
+type MapSearchContextProps = {
+  flyToCenter: ((center: Position) => void) | null
```

Comment:
if there is and will be only one function the context can return this function without wrapping it into an object and we get rid of unnecessary objects both here and in every place used.